### PR TITLE
Source Faker: Enable zero-config execution by making 'count' an optional config param

### PR DESCRIPTION
**Pull request by aaronsteers**
This allows us to use `source-faker` with zero config. Prior to this change, 'count' was required - even though it was defined with the default of 1,000 records.

Drive-by change:

- The 'description' field of 'count' was not 100% correct. I've updated it to correctly describe the current behaviors.

Motivations:

- We use `source-faker` often in PyAirbyte demos and tutorials. For demo and testing purposes, it's helpful to be able to execute the source with zero configuration.

Impact

- This should have zero impact on existing users.

### CODE REVIEW

File: airbyte-integrations/connectors/source-faker/source_faker/spec.json [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
     "title": "Faker Source Spec",
     "type": "object",
-    "required": ["count"],
+    "required": [],
     "additionalProperties": true,
     "properties": {
       "count": {
         "title": "Count",
-        "description": "How many users should be generated in total.  This setting does not apply to the purchases or products stream.",
+        "description": "How many users should be generated in total. By default, one purchase per user will also be created. This setting does not apply to the products stream.",
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-faker/source_faker/spec.json:12]: huh! I thought you got 10 purchases for every 10 users (not evenly distributed).  If you make 1K users by default, you should also get 1K purchases, if the stream is enabled

**Review by evantahler — APPROVED** [review]

File: airbyte-integrations/connectors/source-faker/source_faker/spec.json (hunk 1 above)
Comment (aaronsteers) [reply airbyte-integrations/connectors/source-faker/source_faker/spec.json:12]: Maybe this is true. I've only observed that the number of records of those two streams always matched. I'll double check the logic and will update the description as needed before merging.

Thanks!

**Comment by natikgadzhi** [comment]
And another example where the assortment of formatting tools don't agree. Ruff everything

File: airbyte-integrations/connectors/source-faker/source_faker/spec.json (hunk 1 above)
Comment (aaronsteers) [reply airbyte-integrations/connectors/source-faker/source_faker/spec.json:12]: Thanks, @evantahler, for providing the correct spec on how records are generated. Updated per your comment into the below revised description:

```suggestion
        "description": "How many users should be generated in total. The purchases table will be scaled to match, with 10 purchases created per 10 users. This setting does not apply to the products stream.",
```

**Review by evantahler — APPROVED** [review]

**Comment by evantahler** [comment]
I found the purchase-generating code! https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-faker/source_faker/purchase_generator.py#L51-L66

**Review by alafanechere — APPROVED** [review]
CI is failing because of a new CAT test...
https://github.com/airbytehq/airbyte/pull/36001

**Comment by aaronsteers** [comment]
Is this a new test?

image

**Comment by aaronsteers** [comment]
I just updated to latest CDK version:

`poetry add airbyte-cdk@latest`

**Pull request merged**